test/common/websocket/codec_test.cc

```diff
@@ -0,0 +1,824 @@
+  EXPECT_EQ(0x37fa213d, frames.value()[1].masking_key_.value());
+
+  std::string ping_response_payload;
+  ping_response_payload.resize(5);
+  (*(frames.value()[1].payload_)).copyOut(0, 5, ping_response_payload.data());
+  // Unmasking the ping response payload
+  for (size_t i = 0; i < ping_response_payload.size(); ++i) {
+    ping_response_payload[i] ^=
+        (frames.value()[1].masking_key_.value() >> (8 * (3 - i % 4))) & 0xff;
+  }
+  EXPECT_EQ("Hello", ping_response_payload);
+}
+
+// 256 bytes binary message in a single unmasked frame
+// 0x82 0x7E 0x0100 [256 bytes of binary data]
+TEST(WebSocketCodecTest, decode16BitBinaryUnmaskedFrame) {
+  Buffer::OwnedImpl buffer;
+  Buffer::addSeq(buffer,
+                 {0x82, 0x7e, 0x01, 0x00, 0xd9, 0xac, 0x96, 0x06, 0xc2, 0x83, 0x41, 0x31, 0x16,
+                  0x6c, 0x27, 0x6e, 0x3c, 0xc9, 0x6b, 0x7f, 0x12, 0x32, 0xb4, 0x3f, 0x19, 0x9a,
+                  0x0a, 0x23, 0x6e, 0xfb, 0xb0, 0xda, 0x40, 0x4a, 0xd2, 0xf4, 0xc0, 0xee, 0xa3,
+                  0xc2, 0x3e, 0x7c, 0x96, 0x26, 0x3f, 0x14, 0x34, 0xee, 0xfe, 0xa5, 0x42, 0x23,
+                  0x5b, 0xc8, 0x43, 0xe4, 0xd4, 0xf7, 0xde, 0x79, 0xa6, 0xf9, 0x40, 0xfa, 0xfe,
+                  0xb0, 0x8e, 0xe7, 0x26, 0xa9, 0xa3, 0x6f, 0x07, 0x5c, 0x0a});
+  Buffer::addSeq(buffer,
+                 {0xf1, 0x78, 0xc9, 0xb7, 0x34, 0x80, 0xd9, 0x22, 0xdd, 0x68, 0x0a, 0x34, 0xba,
+                  0x84, 0x64, 0x4d, 0x62, 0x98, 0x44, 0xce, 0x92, 0x0c, 0x16, 0xb1, 0xa5, 0x9b,
+                  0x23, 0xb2, 0xf5, 0x6a, 0x99, 0xc7, 0x9f, 0x89, 0xaf, 0x4b, 0xe8, 0x7a, 0x5e,
+                  0x7d, 0xf5, 0xe6, 0x6a, 0x3e, 0x4a, 0xc9, 0x5e, 0x70, 0xbf, 0x04, 0x3c, 0x33,
+                  0x2a, 0x16, 0xc5, 0xb4, 0xd7, 0x26, 0xa8, 0xca, 0x78, 0xe2, 0x1d, 0xe1, 0xb1,
+                  0x8c, 0xde, 0xa8, 0x35, 0xd7, 0xda, 0x3a, 0x07, 0xb2, 0xf7, 0x71, 0x54, 0x31,
+                  0x7a, 0x1f, 0x45, 0x1c, 0xf9, 0x82, 0xd5, 0x47, 0xfe, 0xc7, 0xb7, 0xd0});
+
+  Buffer::addSeq(buffer, {0x75, 0xe6, 0x2e, 0x56, 0x91, 0x66, 0xb4, 0x89, 0x66, 0xa3, 0xd4, 0xaf,
+                          0x15, 0x3e, 0x7d, 0xa0, 0xd6, 0xca, 0xf9, 0xc7, 0xd9, 0x5d, 0xc6, 0x33,
+                          0x9d, 0xd5, 0x41, 0x22, 0x0f, 0x18, 0xc1, 0xe7, 0xff, 0x8f, 0x2e, 0x6c,
+                          0xaa, 0x81, 0x0f, 0x4e, 0xa5, 0xb6, 0x25, 0x4e, 0x8e, 0x3b, 0x25, 0x1e,
+                          0x14, 0x58, 0x12, 0x5f, 0xfc, 0xe8, 0xb8, 0xc8, 0xc9, 0xd0, 0xa1, 0x53,
+                          0x04, 0xdb, 0x91, 0xd6, 0x9a, 0xc6, 0xa1, 0xe8, 0xd7, 0xb0, 0x49, 0x6d,
+                          0xd8, 0x00, 0x96, 0x41, 0x8d, 0x54, 0xdf, 0x0a, 0x87, 0xea, 0xe7, 0x4c,
+                          0xdf, 0x16, 0x9a, 0x30, 0xa0, 0x26, 0x1c, 0xc4, 0x2c, 0xad, 0xfd});
+
+  Decoder decoder;
+  absl::optional<std::vector<Frame>> frames = decoder.decode(buffer);
+
+  ASSERT_TRUE(frames.has_value());
+  EXPECT_EQ(1, frames.value().size());
+  EXPECT_EQ(2, frames.value()[0].opcode_);
```

Comment:
Here and elsewhere, when checking an opcode, let's use the named constants such as `kFrameOpcodeText`